sos/report/plugins/saltmaster.py

```diff
@@ -26,7 +29,38 @@ def setup(self):
 
         self.add_copy_spec("/etc/salt")
         self.add_forbidden_path("/etc/salt/pki/*/*.pem")
-        self.add_cmd_output("salt-key --list all")
+
+        self.add_copy_spec(["/etc/machine-id",
+                            "/var/lib/dbus/machine-id"])
+
+        self.add_pillar_roots()
+        self.add_cmd_output([
+            "salt-master --version",
+            "systemctl --full status salt-master",
+            "systemctl --full status salt-api",
+            "salt-key --list all",
+            "salt-run jobs.list_jobs --out=yaml",
+            "salt-run manage.list_state --out=yaml",
+            "salt-run manage.list_not_state --out=yaml",
+            "salt-run manage.joined --out=yaml"
+            ], timeout=30)
+
+    def add_pillar_roots(self):
+        cfgs = glob.glob("/etc/salt/master.d/*conf")
+        main_cfg = "/etc/salt/master"
+
+        if os.path.exists(main_cfg):
+            cfgs.append(main_cfg)
```

Comment:
There is a wrapper in `Plugin` for this - `Plugin.path_exists()` which automatically handles a non `/` sysroot, in the event we're running in e.g. a container or recovery environment.

Reply:
Fixed, nice catch!